stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -106,27 +106,82 @@ end
 
 openblas_get_config() = strip(unsafe_string(ccall((@blasfunc(openblas_get_config), libblas), Ptr{UInt8}, () )))
 
+function guess_vendor()
+    # like determine_vendor, but guesses blas in some cases
+    # where determine_vendor returns :unknown
+    ret = vendor()
+    if Sys.isapple() && (ret == :unknown)
+        ret = :osxblas
+    end
+    ret
+end
+
+
 """
-    set_num_threads(n)
+    set_num_threads(n::Integer)
+    set_num_threads(::Nothing)
```

Comment:
This seems like an unusual API, I would prefer `set_num_threads()` instead of `set_num_threads(nothing)`.

Reply:
This is to allow the pattern
```julia
default = BLAS.get_num_threads() # returns nothing on exotic platforms
BLAS.set_num_threads(1)
# do stuff
BLAS.set_num_threads(default)
```

Reply:
That will still work if you make the signature `set_num_threads(n=nothing)`

Reply:
To me allowing `nothing` is a hack to allow the above pattern on strange platforms. It is not a thing I would encourage or that I think needs more convenient syntax.

Reply:
I imagine if you have code like `set_num_threads()` it is more likely you forgot to pass the number of threads, than that you really want to invoce the `nothing` method.

Reply:
I don't think we should define `set_num_threads() = set_num_threads(nothing)`. It would send a wrong message that `set_num_threads(nothing)` is somehow a reasonable default.  But it's not.  It is the last resort that exists only for supporting the rollback use case.

But this is not clear from the current docstring. I think it's better to clarify this.

Reply:
Perhaps clearest to show the pattern which motivates this:
```
Set the number of threads the BLAS library should use.

Also accepts `nothing`, in which case it tries to set set the default number of threads.
On exotic variants of BLAS, `nothing` may be returned by ` get_num_threads()`.
Accepting it means that the following pattern will not give an error:

old = get_num_threads()
set_num_threads(1)
@threads for i in 1:10
    # single-threaded BLAS calls
end
set_num_threads(old)
```

Reply:
@fredrikekre Does the argument here makes sense? It'd be nice if you can have a look at the docstring.